tests/by-util/test_tail.rs

```diff
@@ -4824,3 +4824,40 @@ fn test_obsolete_encoding_windows() {
         .stderr_is("tail: bad argument encoding: '-�b'\n")
         .code_is(1);
 }
+
+#[test]
+fn test_following_with_pid() {
+    use std::process::Command;
+    use std::time::Duration;
+    use std::thread::sleep;
+
+    let scene = TestScenario::new(util_name!());
+
+    let sleep_command = Command::new("sleep")
+        .arg("10")
+        .spawn()
+        .expect("failed to start sleep command");
+
+    let sleep_pid = sleep_command.id();
+    
+    // when -f is specified, tail should die after
+    // the pid from --pid also dies
+    let mut child = scene
+        .ucmd()
+        .args(&["--pid", &sleep_pid.to_string(), "-f"])
+        .set_stdin(Stdio::null())
+        .stderr_to_stdout()
+        .run_no_wait();
+
+    sleep(Duration::from_secs(1));
+
+    child.make_assertion_with_delay(500).is_alive();
+
+    Command::new("kill")
+        .arg("-9")
+        .arg(sleep_pid.to_string())
+        .output()
+        .expect("failed to kill sleep command");
+
```

Comment:
After killing `sleep`, we have a perfect opportunity to quickly check that `tail` really does terminate by itself. Maybe insert something like this?
```
    child.make_assertion_with_delay(1000).is_alive();
```

Reply:
Hey, for checking if it's `is_alive`, why not `is_not_alive`. Running the commands manually, I confirm that the tail exits after sleep exits, and after it's killed. But, I changed it to `is_not_alive`, and it failed on Linux. Although it passed on Windows. I need to switch back over to Windows and confirm it's breaking now

Reply:
To answer my previous question, I believe it _should_ be `is_not_alive`, and the cause for failure was because sleep's parent was the Rust test program, and without exiting, and without reaping, it became a zombie. However, the libc kill function in `is_dead` saw it as still alive (returning `0`), and the reason the program exited before was because of `recv_timeout(settings.sleep_sec)` in watch.rs (currently line 532 in this commit), which I assume errored out, leading to the program ending, and leading to the appearance that it detected the pid specified dying and following suite, when it wasn't actually the case.